src/util.ts

```diff
@@ -192,6 +199,32 @@ export function getRuntimeTrackingString(): string {
   }
 }
 
+export function getDirName() {
```

Comment:
Do you mind testing this with `esbuild`? I'm concerned the fallback logic might break like it broke in a previous PR:
- https://github.com/googleapis/nodejs-storage/issues/1934
- https://github.com/googleapis/nodejs-storage/pull/1892/files#diff-10d5d48c419530b01c345d22a42140d1a726097de84b7525de74f8d5f6fbccc7R619-R625

If it doesn't work perhaps we can have this as a separate .cjs file and import it from both module formats for now.

Reply:
I'll give this a test with esbuild and report back.

Reply:
Added a different helper to fix esbuild problem with package.json.